sdk/nodejs/runtime/resource.ts

```diff
@@ -509,201 +509,217 @@ export function registerResource(
-                        getId: () => undefined,
-                        getObject: () => req.getObject(),
-                        getPropertydependenciesMap: () => undefined,
-                        getResult: () => 0,
-                    };
-                }
 
-                resop.resolveURN(resp.getUrn(), err);
+                    resop.resolveURN(resp.getUrn(), err);
 
-                // Note: 'id || undefined' is intentional.  We intentionally collapse falsy values to
-                // undefined so that later parts of our system don't have to deal with values like 'null'.
-                if (resop.resolveID) {
-                    const id = resp.getId() || undefined;
-                    resop.resolveID(id, id !== undefined, err);
-                }
+                    // Note: 'id || undefined' is intentional.  We intentionally collapse falsy values to
+                    // undefined so that later parts of our system don't have to deal with values like 'null'.
+                    if (resop.resolveID) {
+                        const id = resp.getId() || undefined;
+                        resop.resolveID(id, id !== undefined, err);
+                    }
 
-                const deps: Record<string, Resource[]> = {};
-                const rpcDeps = resp.getPropertydependenciesMap();
-                if (rpcDeps) {
-                    for (const [k, propertyDeps] of resp.getPropertydependenciesMap().entries()) {
-                        const urns = <URN[]>propertyDeps.getUrnsList();
-                        deps[k] = urns.map((urn) => newDependency(urn));
+                    const deps: Record<string, Resource[]> = {};
+                    const rpcDeps = resp.getPropertydependenciesMap();
+                    if (rpcDeps) {
+                        for (const [k, propertyDeps] of resp.getPropertydependenciesMap().entries()) {
+                            const urns = <URN[]>propertyDeps.getUrnsList();
+                            deps[k] = urns.map((urn) => newDependency(urn));
+                        }
                     }
-                }
 
-                // Now resolve the output properties.
-                const keepUnknowns = resp.getResult() !== resproto.Result.SUCCESS;
-                await resolveOutputs(res, t, name, props, resp.getObject(), deps, resop.resolvers, err, keepUnknowns);
+                    // Now resolve the output properties.
+                    const keepUnknowns = resp.getResult() !== resproto.Result.SUCCESS;
+                    await resolveOutputs(
+                        res,
+                        t,
+                        name,
+                        props,
+                        resp.getObject(),
+                        deps,
+                        resop.resolvers,
+                        err,
+                        keepUnknowns,
+                    );
+                    done();
+                });
+            })
+            .catch((err) => {
```

Comment:
The catch here that calls 'done()' is the actual change here, but adding a catch clause causes a reformat of the whole block.